### ISSUE TITLE
How to include '@' (explicit character) in the mention ?

### ISSUE THREAD

**Issue by RahulSDeshpande** [linkedin/Spyglass#60]
Hey guys, great library !! 🥇

I am exploring this lib since 3 days for the user mention feature & I have a requirement to show '@' (explicit character) in the MentionEditText, even after the user selects a suggestion from the list of suggestions.
Here is an example:

Current behaviour:
Hello there, **Rahul Deshpande** here.

Expected behaviour:
Hello there, _explicit character ->_ @**Rahul Deshpande** here.

Having needed this, that explicit character must be spanned with the highlighted colour, etc.. just like the mention & must behave same like it.
**In short, it must be a part of that mention.**

I tried to search in the library, but couldn't find any convenience method for this configuration, so could any please help me ?

💃 Cheers !!

**Comment by nhibner** [comment]
Hey, thanks for the compliment :)

For your use case, there are a couple of ways to approach it that I can think of:

1. You can receive a callback whenever a mention is added/deleted to your text. From this callback, you can modify the text however you would like (in this case, prepending a `@` character at the beginning of the mention). See: https://github.com/linkedin/Spyglass/blob/master/spyglass/src/main/java/com/linkedin/android/spyglass/ui/MentionsEditText.java#L1434

2. Alternatively, you can customize the text that gets inserted for the mention. For example, take a look at the `Person` class from the sample app and how it implements the `Mentionable` interface -- you could have a custom class that includes the `@` symbol as part of the display text for the mention. For reference: https://github.com/linkedin/Spyglass/blob/master/spyglass-sample/src/main/java/com/linkedin/android/spyglass/sample/data/models/Person.java#L37

Hope that helps, please let me know if you run into issues.

**Comment by RahulSDeshpande** [comment]
Hey @nhibner, exactly !!

Yesterday night I exactly tried the same solution as you mention in your 1st point 💃, & its working perfectly, but its an workaround, like this:
```java
@Override
public void onMentionAdded(@NonNull Mentionable mention, @NonNull String text, int start, int end) {
	..
	commentEditText.getText().insert(start, String.valueOf(prefixSymbol), 0, 1);
	CharacterStyle characterStyle = new ForegroundColorSpan(context.getResources().getColor(R.color.colorAccent));
	Spannable spannable = commentEditor.getMentionsText();
	spannable.setSpan(characterStyle, start, start + 1, Spannable.SPAN_EXCLUSIVE_EXCLUSIVE);
	..
}

@Override
public void onMentionDeleted(@NonNull Mentionable mention, @NonNull String text, int start, int end) {
	..
	if (removeTaggedUserNameText(mention.getSuggestiblePrimaryText())) {
		isCommentUpdatedManually = true;
		commentEditor.getText().delete(start - 1, start);
		..
	}
}
```


Where as your 2nd solution though is very interesting & I might give it a try.
But in that case, ```@``` will surely be the part of the mention text, & we need to have the ```.startsWith("@")``` checks all over the code, wherever we get the call back in the ```onMentionAdded()``` & while maintaining the mentions data structure.

Correct me please :)

💃 Cheers !!
